firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java

```diff
@@ -299,6 +301,20 @@ public void testLimitAppliesOrdering() {
     verifyResults(query, "coll/doc1", "coll/doc3");
   }
 
+  @Test
+  public void testIndexEntriesAreUpdated() {
+    SQLiteIndexManager sqLiteIndexManager = (SQLiteIndexManager) indexManager;
+    sqLiteIndexManager.addFieldIndex(
+        fieldIndex("coll", "value", FieldIndex.Segment.Kind.ASCENDING));
+    Query query = query("coll").orderBy(orderBy("value"));
+
+    addDoc("coll/doc1", map("value", true));
+    verifyResults(query, "coll/doc1");
+
+    addDocs(doc("coll/doc1", 1, map()), doc("coll/doc2", 1, map("value", true)));
```

Comment:
Question, is this PR supposed to handle document deletion? What happens if we call `addDocs(deletedDoc("coll/doc2"))` after this line?

Reply:
I added a test for this but this already works as is as deleted docs will not have any fields, and as such they will not have index entries.

Reply:
Thanks, that is what I suspected.